_specifications/specification-3-16.md

```diff
@@ -2779,6 +2779,31 @@ _Response_:
 * result: `any` \| `null`
 * error: code and message set in case an exception happens during the request.
 
+#### <a href="#client_executeCommand" name="client_executeCommand" class="anchor">Execute a client command(:leftwards_arrow_with_hook:)</a>
+
+The `client/executeCommand` request is similar to `workspace/executeCommand` but is sent from the server to the client, to trigger command execution on the client. 
+
+_Client Capability_:
+* property path (optional): `client.executeCommand`
+* property type: `ClientExecuteCommandClientCapabilities` defined as follows:
+
+```typescript
+export interface ClientExecuteCommandClientCapabilities {
+	/**
+	 * The client supports message 'client/executeCommand'.
+	 */
+	supported: boolean;
+}
+```
+
+_Request:_
+* method: 'client/executeCommand'
+* params: `ExecuteCommandParams` defined exactly the same way as for 'workspace/executeCommand'. 
+
+_Response_:
+* result: `any` \| `null`
+* error: code and message set in case an exception happens during the request.
```

Comment:
I would suggest extra wording here:

The client shall return ErrorCode.InvalidParams when the server requests

- a command that doesn't exist
- invalid params for the command signature

Reply:
Agreed. Done!